roadmap/implementers-guide/src/node/validity/approvals.md

```diff
@@ -0,0 +1,49 @@
+# Approval subsystem
+
+The approval subsystem determines whether a relay chain block can be considered for finality.  It does so by running validity checks on the candidates declared available in that relay chain block.  
+
+These approval validity checks differ from the backing validity checks performed before starting availability:
+
+- In backing, adversaries could select when they propose invalid candidates based upon when they control the parachain's backing validators who perform the checks.
+
+- In approvals, we randomly assigns individual validators to check specific candidates without giving adversaries foreknowledge about either which honest validators get assigned to which candidates, or even how many check.  Availability prevents adversaries from choosing which validators obtain their possibly invalid candidate.
+
+As such, approval checks provide significantly more security than backing checks, so polkadot achieves some fixed security level most efficiently when we perform more approval checks per backing check or per relay chain block.  
+
+...
+
+Approval requires two gossiped message types, assignment notices created by its assignments subsystem, and approval votes sent by our approval checks subsystem when authorized by the candidate validity utility subsystem.  
```

Comment:
Oh, these are actually gossiped? Can we bound them somehow? I guess only those where the VRF is low enough & view of ourself and the peer overlaps.

In other cases, we have separate subsystems for networking and core logic. But we can handle the split of this file into individual components after this is merged.

Reply:
Assignment notices should always arrive several seconds before their approval vote. 

Assignment notices are bounded by them being VRF signatures:  Any validator could gossip out maybe num_relay_vrf_modulo + num_cores assignment notices but those num_cores would not be gossiped onwards by other validators until their delay_tranche output by the VRF gets reached, which lies beyond the senders' control.  You could make this more polite if you liked, but not sure that's necessarily. 

Approval votes are always gossiped so yes every validator could send out num_cores approval votes, but doing so sounds mildly risky.